azkaban-exec-server/src/test/java/azkaban/execapp/JobRunnerTest.java

```diff
@@ -124,6 +129,30 @@ public void testBasicRun() throws IOException {
         .assertEvents(EventType.JOB_STARTED, EventType.JOB_STATUS_CHANGED, EventType.JOB_FINISHED);
   }
 
+  private void checkRequiredJobProperties(JobRunner runner, File logFile) {
+    Field jobField = null;
+    try {
+      jobField = runner.getClass().getDeclaredField("job");
```

Comment:
What is the benefit of using reflection here?

Reply:
It prevents us from having to create a get method in JobRunner class to access its private property 'job'. It prevents us from breaking encapsulation for testing purposes.